### ISSUE TITLE
Enable Dragging on only Long Click

### ISSUE THREAD

**Issue by Raminishekhar** [justasm/DragLinearLayout#41]
enabling drag on long click is not available and no callback method for ondragstop.

**Comment by idelder** [comment]
I found a simple solution to this.

1. Add a boolean to DragLinearLayout

`static boolean canDrag = false`

2. In setViewDraggable, add an onLongClickListener to the new view that simply does

`dragHandle.setOnLongClickListener(new OnLongClickListener() {`
`@Override`
`public boolean onLongClick(View view) {`
`canDrag = true;`
`return false;`
 `}`
`});`

3. Replace all instances of `startDrag()` with:

`if (canDrag) {`
`startDrag();`
 `return true;`
`}`

4. In `startDrag()` AND `onTouchEnd()` add 

`canDrag = false`

Basically, if the view being touched has also been long clicked then carry out activities as normal. I also removed the slop check in ACTION_MOVE because it was extremely difficult to do.

Sorry about the formatting. I made an account to write this comment.

**Comment by himanshu1706** [comment]
@idelder I have applied the code you have suggested but drag is not working onLong press.

**Comment by devishell** [comment]
Timer may helps
